resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverOptions.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+/**
+ * Represents options defined in a file of the format <a href=https://linux.die.net/man/5/resolver>etc/resolv.conf</a>.
+ */
+final class UnixResolverOptions {
+
+    private final int ndots;
+    private final int timeout;
+    private final int attempts;
+
+    UnixResolverOptions(int ndots, int timeout, int attempts) {
+        this.ndots = ndots;
+        this.timeout = timeout;
+        this.attempts = attempts;
+    }
+
+    public static UnixResolverOptions.Builder newBuilder() {
+        return new UnixResolverOptions.Builder();
+    }
+
+    /**
+     * The number of dots which must appear in a name before an initial absolute query is made.
+     * The default value is {@code 1}.
+     */
+    public int ndots() {
+        return ndots;
+    }
+
+    /**
+     * The timeout of each DNS query performed by this resolver (in seconds).
+     * The default value is {@code 5}.
+     */
+    public int timeout() {
+        return timeout;
+    }
+
+    /**
+     * The maximum allowed number of DNS queries to send when resolving a host name.
+     * The default value is {@code 16}.
+     */
+    public int attempts() {
+        return attempts;
+    }
+
+    public static final class Builder {
+
+        private int ndots = 1;
+        private int timeout = 5;
+        private int attempts = 16;
+
+        private Builder() {
+        }
+
+        public void setNdots(Integer ndots) {
```

Comment:
why not use `int` ?

Reply:
`null` values are used for error control when parsing from strings failed for example (and not override defaults).
But I see you don't like it (I'm not fond of it either), so I moved catching `NumberFormatException` from `parseResIntOption` to `parseResOption`.